src/EnergyPlus/UnitVentilator.cc

```diff
@@ -692,108 +554,92 @@ namespace UnitVentilator {
                     ErrorsFound = true;
                 }
             }
-            //   A13, \field Coil Option
-            //        \required-field
-            //        \type choice
-            //        \key None
-            //        \key Heating
-            //        \key Cooling
-            //        \key HeatingAndCooling
 
             {
-                auto const SELECT_CASE_var(Alphas(13));
-                if (SELECT_CASE_var == "HEATINGANDCOOLING") {
-                    state.dataUnitVentilators->UnitVent(UnitVentNum).CoilOption = BothOption;
-                } else if (SELECT_CASE_var == "HEATING") {
-                    state.dataUnitVentilators->UnitVent(UnitVentNum).CoilOption = HeatingOption;
-                } else if (SELECT_CASE_var == "COOLING") {
-                    state.dataUnitVentilators->UnitVent(UnitVentNum).CoilOption = CoolingOption;
-                } else if (SELECT_CASE_var == "NONE") {
-                    state.dataUnitVentilators->UnitVent(UnitVentNum).CoilOption = NoneOption;
+                auto const coilOption(Alphas(13));
+                if (coilOption == "HEATINGANDCOOLING") {
+                    unitVent.CoilOption = CoilOption::Both;
+                } else if (coilOption == "HEATING") {
+                    unitVent.CoilOption = CoilOption::Heating;
+                } else if (coilOption == "COOLING") {
+                    unitVent.CoilOption = CoilOption::Cooling;
+                } else if (coilOption == "NONE") {
+                    unitVent.CoilOption = CoilOption::None;
                 } else {
-                    ShowSevereError(state,
-                                    std::string{RoutineName} + CurrentModuleObject + "=\"" + state.dataUnitVentilators->UnitVent(UnitVentNum).Name +
-                                        "\", invalid");
+                    ShowSevereError(state, std::string{RoutineName} + CurrentModuleObject + "=\"" + unitVent.Name + "\", invalid");
                     ShowContinueError(state, "illegal value: " + cAlphaFields(13) + "=\"" + Alphas(13) + "\".");
                     ErrorsFound = true;
                 }
             }
 
-            state.dataUnitVentilators->UnitVent(UnitVentNum).FanSchedPtr = GetScheduleIndex(state, Alphas(14));
+            unitVent.FanSchedPtr = ScheduleManager::GetScheduleIndex(state, Alphas(14));
             // Default to cycling fan when fan mode schedule is not present
-            if (!lAlphaBlanks(14) && state.dataUnitVentilators->UnitVent(UnitVentNum).FanSchedPtr == 0) {
-                ShowSevereError(state,
-                                CurrentModuleObject + " \"" + state.dataUnitVentilators->UnitVent(UnitVentNum).Name + "\" " + cAlphaFields(14) +
-                                    " not found: " + Alphas(14));
+            if (!lAlphaBlanks(14) && unitVent.FanSchedPtr == 0) {
```

Comment:
I'm not a huge fan of this idiom where we first try to find something (e.g., `FanSchedPtr`) and only then check if the thing we are trying to find is blank. Feels very backwards. Feels like we should check if the thing is blank first and only if it isn't, do the search for it.